sticker/src/model/mod.rs

```diff
@@ -0,0 +1,278 @@
+    ///
+    /// * `attention_mask`: specifies which sequence elements should
+    ///    be masked when applying the encoder.
+    /// * `token_mask`: specifies which sequence elements should be
+    ///    masked when computing the loss. Typically, this is used
+    ///    to exclude padding and continuation word pieces.
+    /// * `targets`: the labels to be predicted, per encoder name.
+    /// * `label_smoothing`: apply label smoothing, redistributing
+    ///   the given probability to non-target labels.
+    /// * `train`: indicates whether this forward pass will be used
+    ///   for backpropagation.
+    /// * `freeze_embeddings`: exclude embeddings from backpropagation.
+    /// * `freeze_encoder`: exclude the encoder from backpropagation.
+    #[allow(clippy::too_many_arguments)]
+    pub fn loss(
+        &self,
+        inputs: &Tensor,
+        attention_mask: &Tensor,
+        token_mask: &Tensor,
+        targets: &HashMap<String, Tensor>,
+        label_smoothing: Option<f64>,
+        train: bool,
+        freeze_embeddings: bool,
+        freeze_encoder: bool,
+    ) -> (Tensor, HashMap<String, Tensor>, HashMap<String, Tensor>) {
+        let encoding = self.encode(
+            inputs,
+            attention_mask,
+            train,
+            freeze_embeddings,
+            freeze_encoder,
+        );
+
+        let token_mask = token_mask.to_kind(Kind::Float);
+        let token_mask_sum = token_mask.sum(Kind::Float);
+
+        let mut encoder_specific_loss = HashMap::with_capacity(self.classifiers.len());
+        let mut encoder_specific_accuracy = HashMap::with_capacity(self.classifiers.len());
+        for (encoder_name, classifier) in &self.classifiers {
+            let (loss, correct) =
+                classifier.losses(&encoding, &targets[encoder_name], label_smoothing, train);
+            let loss = (loss * &token_mask).sum(Kind::Float) / &token_mask_sum;
+            let acc = (correct * &token_mask).sum(Kind::Float) / &token_mask_sum;
+
+            encoder_specific_loss.insert(encoder_name.clone(), loss);
+            encoder_specific_accuracy.insert(encoder_name.clone(), acc);
+        }
+
+        let summed_loss = encoder_specific_loss.values().fold(
+            Tensor::zeros(&[], (Kind::Float, inputs.device())),
+            |summed_loss, loss| summed_loss + loss,
+        );
+
+        (
+            summed_loss,
+            encoder_specific_loss,
+            encoder_specific_accuracy,
+        )
+    }
+
+    /// Do a forward pass of the model on the inputs.
+    ///
+    /// * `attention_mask`: specifies which sequence elements should
+    ///    be masked when applying the encoder.
+    /// * `train`: indicates whether this forward pass will be used
+    ///   for backpropagation.
+    /// * `freeze_embeddings`: exclude embeddings from backpropagation.
+    /// * `freeze_encoder`: exclude the encoder from backpropagation.
+    pub fn forward_t(
```

Comment:
is `forward_t` used for training?

Reply:
It's currently not used at all, so I have removed it from this PR.